scenarios/mixed_arc_must_get_agent_activity/src/main.rs

```diff
@@ -0,0 +1,288 @@
+
+    if should_record {
+        let app_client = ctx.get().app_client();
+        let network_stats = ctx
+            .runner_context()
+            .executor()
+            .execute_in_place(async move { Ok(app_client.dump_network_stats().await?) })?;
+
+        let metric = ReportMetric::new("mixed_arc_must_get_agent_activity_open_connections")
+            .with_tag("behaviour", ctx.assigned_behaviour().to_string())
+            .with_field("value", network_stats.connections.len() as u32);
+        ctx.runner_context().reporter().clone().add_custom(metric);
+
+        ctx.get_mut().scenario_values.open_connections_last_recorded = Some(now);
+    }
+
+    Ok(())
+}
+
+fn agent_behaviour_write(
+    ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext<ScenarioValues>>,
+) -> HookResult {
+    record_open_connections_if_necessary(ctx)?;
+
+    let batch_count = ctx.get().scenario_values.batch_count;
+
+    // Create a batch of 10 entries
+    let _: () = call_zome(
+        ctx,
+        "validated_must_get_agent_activity",
+        "create_sample_entries_batch",
+        CreateEntriesBatchInput {
+            num_entries: BATCH_SIZE,
+            batch_num: batch_count,
+        },
+    )?;
+
+    ctx.get_mut().scenario_values.batch_count += 1;
+
+    let agent_pub_key = ctx.get().cell_id().agent_pubkey().to_string();
+    let metric = ReportMetric::new("mixed_arc_must_get_agent_activity_entry_created_count")
+        .with_tag("agent", agent_pub_key)
+        .with_tag("behaviour", ctx.assigned_behaviour().to_string())
+        .with_field("value", ctx.get().scenario_values.batch_count * BATCH_SIZE);
+    ctx.runner_context().reporter().clone().add_custom(metric);
+
+    Ok(())
+}
+
+fn agent_behaviour_must_get_agent_activity(
+    ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext<ScenarioValues>>,
+) -> HookResult {
+    record_open_connections_if_necessary(ctx)?;
+
+    let reporter = ctx.runner_context().reporter();
+
+    match ctx.get().scenario_values.write_peer.clone() {
+        Some(write_peer) => {
+            // If we can move on with the next batch, fetch the associated BatchChainTop
+            // ---
+            // Iteration 0: currently_pending_batch_chain_top is None, last_successfully_fetched_batch is None
+            // Iteration 1: currently_pending_batch_chain_top is Some({ batch_num: 0, ...}), last_successfully_fetched_batch is None
+            // Iteration 2.1: currently_pending_batch_chain_top is Some({ batch_num: 0, ...}), last_successfully_fetched_batch is Some(0)
+            // Iteration 2.x: currently_pending_batch_chain_top is Some({ batch_num: 1, ...}), last_successfully_fetched_batch is Some(0)
+            // Iteration 3.3: currently_pending_batch_chain_top is Some({ batch_num: 1, ...}), last_successfully_fetched_batch is Some(1)
+            // Iteration 3.x: currently_pending_batch_chain_top is Some({ batch_num: 2, ...}), last_successfully_fetched_batch is Some(1)
```

Comment:
I tried to make the logic easier to follow with above inline comment - not sure whether I succeeded.

Reply:
I appreciate the attempt but I do not find this easier to follow 😄 . I would prefer prose.